From ae045dca857c1c91af4c6ff283e1075d5e9610ff Mon Sep 17 00:00:00 2001
From: MengLi <meng.li@windriver.com>
Date: Tue, 5 Nov 2019 11:27:16 +0800
Subject: [PATCH 148/151] arch: arm64: dts: add pcie dts file for stratix 10
 platform

Refer to below link about stratix 10 PCIe description,
https://rocketboards.org/foswiki/Projects/Stratix10PCIeRootPortWithMSI
add pcie dts file for stratix 10 platform.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 arch/arm64/boot/dts/altera/Makefile           |  2 +-
 .../altera/socfpga_stratix10_socdk_pcie.dts   | 59 +++++++++++++++++++
 2 files changed, 60 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/altera/socfpga_stratix10_socdk_pcie.dts

diff --git a/arch/arm64/boot/dts/altera/Makefile b/arch/arm64/boot/dts/altera/Makefile
index 9e50a326dff9..9e74497e8729 100644
--- a/arch/arm64/boot/dts/altera/Makefile
+++ b/arch/arm64/boot/dts/altera/Makefile
@@ -1,4 +1,4 @@
 # SPDX-License-Identifier: GPL-2.0-only
-dtb-$(CONFIG_ARCH_STRATIX10) += socfpga_stratix10_socdk.dtb socfpga_stratix10_fpga_update.dtb \
+dtb-$(CONFIG_ARCH_STRATIX10) += socfpga_stratix10_socdk.dtb socfpga_stratix10_fpga_update.dtb socfpga_stratix10_socdk_pcie.dtb \
 				socfpga_stratix10_socdk_nand.dtb
 dtb-$(CONFIG_ARCH_STRATIX10SWVP) += stratix10_swvp.dtb
diff --git a/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk_pcie.dts b/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk_pcie.dts
new file mode 100644
index 000000000000..1d84c45da68f
--- /dev/null
+++ b/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk_pcie.dts
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * Copyright Altera Corporation (C) 2019. All rights reserved.
+ */
+
+#include "socfpga_stratix10_socdk.dts"
+
+/ {
+	soc {
+		s10_hps_bridges: bridge@80000000 {
+			compatible = "simple-bus";
+			reg = <0x80000000 0x20200000>,
+				<0xf9000000 0x00100000>;
+			reg-names = "axi_h2f", "axi_h2f_lw";
+			#address-cells = <2>;
+			#size-cells = <1>;
+			ranges = <0x00000000 0x00000000 0x80000000 0x00040000>,
+				<0x00000000 0x10000000 0x90000000 0x08000000>,
+				<0x00000000 0x20000000 0xa0000000 0x00200000>,
+				<0x00000001 0x00010000 0xf9010000 0x00008000>,
+				<0x00000001 0x00018000 0xf9018000 0x00000080>,
+				<0x00000001 0x00018080 0xf9018080 0x00000010>;
+
+			pcie_0_pcie_s10: pcie@200000000 {
+				compatible = "altr,pcie-root-port-2.0";
+				reg = 	<0x00000000 0x20000000 0x00200000>,
+					<0x00000000 0x10000000 0x10000000>,
+					<0x00000001 0x00010000 0x00008000>;
+				reg-names = "Hip", "Txs", "Cra";
+				interrupt-parent = <&intc>;
+				interrupts = <0 20 4>;
+				interrupt-controller;
+				#interrupt-cells = <1>;
+				device_type = "pci";    /* embeddedsw.dts.params.device_type type STRING */
+				bus-range = <0x00000000 0x000000ff>;
+				ranges = <0x82000000 0x00000000 0x00000000 0x00000000 0x10000000 0x00000000 0x08000000>;
+				msi-parent = <&pcie_0_msi_irq>;
+				#address-cells = <3>;
+				#size-cells = <2>;
+				interrupt-map-mask = <0 0 0 7>;
+				interrupt-map = <0 0 0 1 &pcie_0_pcie_s10 1>,
+						<0 0 0 2 &pcie_0_pcie_s10 2>,
+						<0 0 0 3 &pcie_0_pcie_s10 3>,
+						<0 0 0 4 &pcie_0_pcie_s10 4>;
+			}; //end pcie@0x010000000 (pcie_0_pcie_s10)
+
+			pcie_0_msi_irq: msi@10008080 {
+				compatible = "altr,msi-1.0";
+				reg = <0x00000001 0x00018080 0x00000010>,
+					<0x00000001 0x00018000 0x00000080>;
+				reg-names = "csr", "vector_slave";
+				interrupt-parent = <&intc>;
+				interrupts = <0 19 4>;
+				msi-controller = <1>;   /* embeddedsw.dts.params.msi-controller type NUMBER */
+				num-vectors = <32>;     /* embeddedsw.dts.params.num-vectors type NUMBER */
+			}; //end msi@0x100008000 (pcie_0_msi_irq)
+		};
+	};
+};
-- 
2.26.1

